### DISCUSSION TITLE
About heavy  simulation with OpenOil

### DISCUSSION THREAD

**Discussion by fnamorim** [OpenDrift/opendrift#1065] [General]
Hy everyone,

I am using OpenOil to simulate particles that are subject to weathering like oil, but do not emulsify, and some are heavier than the ocean water, with density of 1620 kg/m3. These heavy particles do not sink, or pass through the mixed layer, when released from surface. Also, they do not sink in a case of a seafloor blowout, instead go directly to the surface. I'd like to know if it is the normal behavior, or if there are some limitations to oil density in the OpenOil formulation. Hope I was clear about my questioning.

Thank you all.

**Comment by knutfrode** [comment]
Hi, and sorry for the late reply,

There is a hidden possibility in OpenDrift to set a custom oiltype:
https://opendrift.github.io/autoapi/opendrift/models/openoil/index.html#opendrift.models.openoil.OpenOil.set_oiltype_by_json
However, I made a quick test, and it seems that ADIOS (which OpenDrift is using for weathering) does not support oils heavier than water.

If you do not really need to calculate weathering changes during the simulation, you could also use the simpler OceanDrift module, and specifying terminal velocities based on your given densities and corresponding particle/droplet radii:
https://opendrift.github.io/gallery/example_vertical_mixing.html

In any case: particles heavier than water (negative buoyancy) will generally sink, but vertical turbulence can still give some intermittent rising, as illustrated with above example.

**Reply by fnamorim** [reply]
Hi Dr. Dagestad,

Thank you for your reply.

I am working now with HNS (Hazardous Noxious Substances) modelling in deep-sea areas close to mining spots, to understand how an accident could affect the deep sea communities. There is scarce material available about this subject, what I found in the available literature is that HNS behaves like oil in the sea, regarding weathering processes and dispersion, but it does not emulsify. There are some HNS considered as a sinker, with density of 1620 kg.m-3. Based on those assumptions I decided to use OpenOil, the problem is the sinker HNS do not sink in a surface spill and go straight to the surface in a spill close to seabed. I was able to build a .json file with the HNS properties, which OpenOil is reading properly. I performed a number of tests and the "settling velocity" formulation adopted by the model is also working properly, the velocities are calculated based on the T,S parameters read from the ROMS model and the droplet diameter range adopted, which is important since the T,S change through depth. Regarding the OpenOil, I decided to set the "deactivate" option in case of reaching seafloor or any barrier (like the seamounts), and in the case of deep spill it worked. I also performed some tests with heavy oils and the results were the same.
Based on your answer, I performed some tests using OceanDrift, calculating an average settling velocity, based on the same formulation adopetd by OpenOil. The droplets keep in the deep depths, in my understanding, because the settling velocity does not change based on T,S change, and also the droplets diameter, thikness and mass.
I am sorry for the long explanation, it was to say that I did as many tests I could, with attention to the details I have in hand. Based on that, and with opinions of some collegues, I decided to use OpenOil and perform only a deep sea spill. Since this type of substances (HNS) is somehow new, but with a lot of attention nowadays, I need to be careful with the assumptions I made. I would appreciate having an opinion from your side. Maybe ADIOS starts to include these substances in their lybrary and in the weathering processes formulation.

Best regards.

**Comment by knutfrode** [comment]
Ok, that is an interesting application. This should be possible to do with OpenDrift, given proper configuration. However, OpenOil is limited to oils, unfortunately not heavier than water, and OceanDrift does not have any weathering functionality.

Ideally you could build your own module subclassing OceanDrift (like OpenOil does), and in the update method you could implement the relevant processes, i.e. changing particle properties (radii, densities, ) based on provided forcing data (temperatures, densities...). 
You could then perhaps implement the method `update_terminal_velocity` which is a placeholder in OceanDrift to be implemented in sub modules for specific applications (e.g. OpenOil, LarvalFish, ...):
https://opendrift.github.io/_modules/opendrift/models/oceandrift.html#OceanDrift.update_terminal_velocity
https://opendrift.github.io/_modules/opendrift/models/openoil/openoil.html#OpenOil.update_terminal_velocity
https://opendrift.github.io/_modules/opendrift/models/larvalfish.html#LarvalFish.update_terminal_velocity

**Comment by fnamorim** [comment]
Hi,

I decided to adopt the OceanDrift, since the weathering processes are not
activated in a deep spill. I managed to update the settling velocity using
the same formulation adopted in OpenOil.
In my opinion, this approach makes more sense.

Thank you for your suggestions.

Fabiola.

On Wed, Mar 8, 2023 at 12:17 PM Knut-Frode Dagestad <
***@***.***> wrote:

> Ok, that is an interesting application. This should be possible to do with
> OpenDrift, given proper configuration. However, OpenOil is limited to oils,
> unfortunately not heavier than water, and OceanDrift does not have any
> weathering functionality.
>
> Ideally you could build your own module subclassing OceanDrift (like
> OpenOil does), and in the update method you could implement the relevant
> processes, i.e. changing particle properties (radii, densities, ) based on
> provided forcing data (temperatures, densities...).
> You could then perhaps implement the method update_terminal_velocity
> which is a placeholder in OceanDrift to be implemented in sub modules for
> specific applications (e.g. OpenOil, LarvalFish, ...):
>
> https://opendrift.github.io/_modules/opendrift/models/oceandrift.html#OceanDrift.update_terminal_velocity
>
> https://opendrift.github.io/_modules/opendrift/models/openoil/openoil.html#OpenOil.update_terminal_velocity
>
> https://opendrift.github.io/_modules/opendrift/models/larvalfish.html#LarvalFish.update_terminal_velocity
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenDrift/opendrift/discussions/1065#discussioncomment-5241077>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMYKKDQUEJBHWOE7ZTZDPATW3B2HLANCNFSM6AAAAAAVJUPBDM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
¸.·´(`_¸.·´(`_¸.·´(`_¸.·´(`_¸.·´(`_¸.·´(`_¸.·´(`_¸.·´(`_¸.·´(`_¸.·´(`_¸.·´(`
_¸.·´(`_¸.·´(`_¸.·´(`_¸.·´(`_¸.·´(`_¸.·´(`
-----------------------------------------------------------------------------------------------------------------
Fabiola Negreiros de Amorim, D.Sc.
*FCUP* -  Faculdade de Ciências da Universidade do Porto
*CIIMAR* - Interdisciplinary Centre of Marine and Environmental Research

*REMO - *Oceanographic Modelling and Observation Network
*GOAT - *Tropical Oceanography Group, UFBA
*LABHIDRO* - Hydraulic Laboratory, Environmental Engineer Dep. - UFES
-----------------------------------------------------------------------------------------------------------------
"Todos estes que aí estão atravancando o meu caminho, Eles passarão! Eu
passarinho..."
 Mario Quintana.

**Comment by knutfrode** [comment]
Very good.
Just be aware that OpenDrift is not very much used/tested for applications with negative buoyancy, compared to positive or neutral buoyancy.
So please report if you experience any unexpected or unphysical behaviour.

The general method `vertical_mixing`from OceanDrift is used by all 3D modules (oil, fish eggs, sediments, microplastics), but where specific methods for terminal velocity and interaction with surface or bottom is implemented in each specific module, and is called at each short sub-timestep of `vertical_mixing`.
https://opendrift.github.io/_modules/opendrift/models/oceandrift.html#OceanDrift.vertical_mixing
The method `vertical_mixing` should probably me more accurately renamed to `vertical_mixing_and_buoyancy`
By combining mixing and buoyancy within the same loop with small timestep (~seconds), one can have a larger time step (~hours) for the advection.

See SedimentDrift module for a minimalistic case of defining `bottom_interaction`, although this is not very much used/tested:
https://opendrift.github.io/_modules/opendrift/models/sedimentdrift.html#SedimentDrift.bottom_interaction

Another example/illustration that might be relevant for your application:
https://opendrift.github.io/gallery/example_seafloor_interaction.html
